### PR TITLE
Fix getting the avatar of a non-existing user

### LINKED ISSUE #14870: getDisplayName() on a non-object when requesting avatar of non-existing user
```
{"reqId":"JY7PzrQi6MGXuOhmAjrQ","remoteAddr":"127.0.0.1","app":"PHP","message":"Call to a member function getDisplayName() on a non-object at \/home\/nickv\/ownCloud\/Master\/core\/avatar\/avatarcontroller.php#106","level":3,"time":"2015-03-13T11:00:01+00:00"}
```

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #14870 

@LukasReschke @rullzer

### CODE REVIEW

**Comment by LukasReschke** [comment]
Related https://github.com/owncloud/core/pull/14874

File: core/avatar/avatarcontroller.php [hunk 1]
```diff
@@ -101,11 +101,13 @@ public function getAvatar($userId, $size) {
 				['Content-Type' => $image->mimeType()]);
 			$resp->setETag(crc32($image->data()));
 		} else {
-			$resp = new DataResponse(
-						['data' => [
-								'displayname' => $this->userManager->get($userId)->getDisplayName()
-							]
-						]);
+			$user = $this->userManager->get($userId);
+			$userName = $user ? $user->getDisplayName() : $userId;
```
Comment (nickvergessen) [inline core/avatar/avatarcontroller.php]: Should be `: ''` so the JS code is right again:
```
					if (result.data && result.data.displayname) {
```

File: tests/core/avatar/avatarcontrollertest.php [hunk 2]
```diff
@@ -143,7 +143,24 @@ public function testGetAvatar() {
...
+		$this->assertEquals(crc32($response->getData()), $response->getEtag());
+	}
+
+	/**
+	 * Fetch the avatar of a non-existing user
+	 */
+	public function testGetAvatarNoUser() {
+		$image = new Image(OC::$SERVERROOT.'/tests/data/testimage.jpg');
+		$this->avatarMock->method('get')->willReturn($image);
+		$this->container['AvatarManager']->method('getAvatar')->willReturn($this->avatarMock);
```
Comment (rullzer) [inline tests/core/avatar/avatarcontrollertest.php]: This does not make sense. If the user does not exists, there will be no avatar to return

Comment (rullzer) [inline tests/core/avatar/avatarcontrollertest.php]: I guess it is just a artifact from a copy of the test function :)

**Comment by rullzer** [comment]
Thanks for fixing my argument mixup!

File: tests/core/avatar/avatarcontrollertest.php (hunk 2 above)
Comment (nickvergessen) [inline tests/core/avatar/avatarcontrollertest.php]: jup, you got me :)

Comment (nickvergessen) [inline tests/core/avatar/avatarcontrollertest.php]: fixed

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by rullzer** [comment]
:+1:

**Pull request merged**